File: pi.js
const diameter = 10000

console.log("Calculate PI by precision of " + diameter*diameter+  " tries")

let dots_in_circle = 0
let dots_total = 0
let center = diameter/2
let radius = diameter/2

for (let i=0; i<diameter; i++) {
    for (let j=0; j< diameter; j++) {
        dots_total++

        let x_distance_from_center = center-i
        if (x_distance_from_center < 0) {
            x_distance_from_center *= -1
        }
        let y_distance_from_center = center-j
        if (y_distance_from_center < 0) {
            y_distance_from_center *= -1
        }

        let distance_from_center = Math.sqrt(x_distance_from_center*x_distance_from_center+y_distance_from_center*y_distance_from_center)
        if (distance_from_center <= radius) {
            dots_in_circle++
        }
    }
}

console.log(dots_in_circle+", "+dots_total)
console.log(4*(dots_in_circle/dots_total))

File: pi.py
import math
# import argparse

# So box is gonna be 100x100 and radius of circle is going to be diameter/2 is 50
# parser = argparse.ArgumentParser()
# parser.add_argument('--weight', default=1000, help="this number pow2 is the number of dots to check. Higher makes things slower but more precise", type=int)
# args = parser.parse_args()
diameter = 10000

print("Calculate PI by precision of " + str(diameter*diameter)+  " tries")

dots_in_circle = 0
dots_total = 0
center = diameter/2
radius = diameter/2

for i in range(0,diameter):
    for j in range(0,diameter):
        dots_total += 1
        # I am working on X, Y, use pythagoras to calculate the distance from center.
        x_distance_from_center = center-i
        if (x_distance_from_center < 0):
            # make the number positive
            x_distance_from_center *= -1
        y_distance_from_center = center-j
        if (y_distance_from_center <0 ):
            y_distance_from_center *= -1
        
        distance_from_center = math.sqrt(x_distance_from_center*x_distance_from_center + y_distance_from_center * y_distance_from_center)
        if (distance_from_center <= radius):
            dots_in_circle +=1 
        

print(dots_in_circle, dots_total)
print(4*(float(dots_in_circle)/float(dots_total)))
